test/ci/utils.sh

```diff
@@ -314,22 +347,56 @@ generate_certs() {
   done
   generate_cert "${device_ca_key}" "${device_ca_crt}" "${device_ca_pub}" "${device_ca_subj}"
   ls -l "${certs_dir}"
+  # Generate HTTPS transport certs (PEM) for services using HTTPS
+  for service in "${services[@]}"; do
+    local proto_var="${service}_protocol"
+    # Safely read protocol with set -u
+    local proto_val="${!proto_var-}"
+    if [ "${proto_val}" = "https" ]; then
+      # Build var names and safely dereference
+      local key_var="${service}_https_key"
+      local crt_var="${service}_https_crt"
+      local subj_var="${service}_https_subj"
+      local key_path="${!key_var-}"
+      local crt_path="${!crt_var-}"
+      local https_subj="/C=US/O=FDO/CN=${service}"
+      if [[ -v ${subj_var} ]]; then
+        https_subj="${!subj_var}"
+      fi
+      # No pub file needed for HTTPS, pass empty third arg
+      generate_cert "${key_path}" "${crt_path}" "" "${https_subj}" pem
+
+      # Ensure ownership and permissions for HTTPS certs/keys
+      local user="go-fdo-server-${service}"
+      local group="go-fdo-server"
+      sudo chown "${user}:${group}" "${key_path}" "${crt_path}" || true
+      sudo chmod 640 "${key_path}" || true
+      sudo chmod 644 "${crt_path}" || true
+
+      # If systemd sysconfig exists, configure ADDITIONAL_OPTS to use these certs
+      local sysconfig_file="/etc/sysconfig/go-fdo-server-${service}"
+      if [[ -f "${sysconfig_file}" && -n "${crt_path}" && -n "${key_path}" ]]; then
+        sudo sed -i "s|^ADDITIONAL_OPTS=\"\"|ADDITIONAL_OPTS=\"--server-cert-path=${crt_path} --server-key-path=${key_path}\"|" "${sysconfig_file}"
+      fi
```

Comment:
This is specific to tmt (rpm/systemd) tests so it should go in `test/fmf/tests/utils.sh`. I would create a `configure_services` function there to generate the `/etc/sysconfig` files depending on the env variables and I will call it from `test/fmf/tests/test-onboarding.sh` so you don't need to add it it `test/fmf/tests/test-onboarding-https.sh`